build_rules.gradle

```diff
@@ -837,6 +837,179 @@ ext.applyJavaNature = {
+
+  // Task for running integration tests
+  task integrationTest(type: Test) {
+    include "**/*IT.class"
+    systemProperties.beamTestPipelineOptions = configuration.integrationTestPipelineOptions
+  }
+}
+
+// When applied in a module's build.gradle file, this closure adds task providing
+// additional dependencies that might be needed while running integration tests.
+ext.provideIntegrationTestingDependencies = {
+  println "provideIntegrationTestingDependencies with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  // See: http://groovy-lang.org/closures.html#implicit-it
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Add runners needed to run integration tests on
+  task packageIntegrationTests(type: Jar) {
+    def runner = configuration.runner
+    def filesystem = configuration.filesystem
+
+    dependencies {
+
+      /* include dependencies required by runners */
+      if (runner?.contains('dataflow')) {
+        testCompile project(path: ":beam-runners-google-cloud-dataflow-java", configuration: 'shadowTest')
+      }
+
+      if (runner?.contains('direct')) {
+        testCompile project(path: ":beam-runners-direct-java", configuration: 'shadowTest')
+      }
+
+      /* include dependencies required by filesystems */
+      if (filesystem?.contains('hdfs')) {
+        testCompile project(path: ":beam-sdks-java-io-hadoop-file-system", configuration: 'shadowTest')
+        shadowTest library.java.hadoop_client
+      }
+    }
+  }
+}
+
+// When applied in a module's build gradle file, this closure provides a task
+// that will involve PerfKitBenchmarker for running integrationTests.
+ext.createPerformanceTestHarness = {
+  println "createPerformanceTestHarness with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  // See: http://groovy-lang.org/closures.html#implicit-it
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // This task runs PerfKitBenchmarker, which does benchmarking of the IO ITs.
+  // The arguments passed to it allows it to invoke gradle again with the desired benchmark.
+  //
+  // To invoke this, run:
+  //
+  // ./gradlew performanceTest \
+  //  -DpkbLocation="<path to pkb.py>"
+  //  -DintegrationTestPipelineOptions='["--numberOfRecords=1000", "<more options>"]' \
+  //  -DintegrationTest=<io test, eg. org.apache.beam.sdk.io.text.TextIOIT> \
+  //  -DitModule=<directory containing desired test, eg. sdks/java/io/file-based-io-tests> \
+  //  -DintegrationTestRunner=direct/dataflow/<other>
```

Comment:
<!--new_thread; commit:3165e6e7301ddfb1565b1de36397bfac474d253b; resolved:0-->
What's the value in providing multiple values for 'runner' parameter. Don't we always run a single execution using a single runner ?

Reply:
There is no value at all - it's only my inadvertence. :) Thanks to your comment I realised this may be misguiding. I'll rephrase this.